cmd/root.go

```diff
@@ -375,17 +362,15 @@ func init() {
 	_ = rootCmd.PersistentFlags().MarkHidden("memory-profile")
 }
 
-const versionMetadataUrl = "https://azcopyvnextrelease.z22.web.core.windows.net/releasemetadata/latest_version.txt"
-
 // always spins up a new goroutine, because sometimes the aka.ms URL can't be reached (e.g. a constrained environment where
 // aka.ms is not resolvable to a reachable IP address). In such cases, this routine will run for ever, and the caller should
 // just give up on it.
 // We spin up the GR here, not in the caller, so that the need to use a separate GC can never be forgotten
 // (if do it synchronously, and can't resolve URL, this blocks caller for ever)
-func beginDetectNewVersion() chan struct{} {
+func beginDetectNewVersion() (chan struct{}, string) {
```

Comment:
I dont think you need to make this return common.AzCopyVersion. Can't you just return that in the version method above?